### ISSUE TITLE
Starting from 0.12.0: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application.

### ISSUE THREAD

**Issue by mahsa2** [tonsky/rum#224]
I get a warning as of `0.12.0+` where if a local variable is updated when its component is unmounted, there is a state update attempt where it causes a warning of **memory leak**. This warning doesn't appear on the previous versions (I tested it on `0.11.5` and `0.11.4`).

[Steps to reproduce]
You can run the following component, then unmount it, and wait for the `setTimeout` call to see the warning.
```clojure
(rum/defcs Layout
  < (rum/local false :*flag)
  [{:keys [*flag]}]
  [:div #?(:cljs (js/setTimeout
                   (fn []
                     (println "setTimeout got called")
                     (reset! *flag true))
                   5000))])
```

Thank you in advance :)

**Comment by roman01la** [comment]
How's that Rum's issue? It's the issue in the code itself, React just warns you to not do that.

**Comment by mahsa2** [comment]
why didn't previous versions have this warning? is it a new warning in the newer react versions?

**Comment by roman01la** [comment]
Ok, I see what's happening here. In `0.12.0` we removed Rum's custom scheduling mechanism, which was based on `requestAnimationFrame`. There was a check for unmounted component, which was needed, because what scheduled is out of sync with React itself, thus some updates could be executed by the time when a component is unmounted.

Now that we removed a custom scheduler and now rely on React's update scheduling mechanism, we don't this check anymore. I consider this to be an improvement, because now React can actually tell you that there's something wrong with your code.

Thanks for bringing this up!
